### PR TITLE
refactor(console): update plan comparison table

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Update plan comparison table accroding to the new pricing design.

The plan table content changes:
- Change "mfa", "sso" and "organizations" feature flags to "Contact" for enterprise plan, since these features will be add-on features in the future.
- Add payment tool tips in the plan comparison table
- Add token limit data on the front-end and make it displayed in the "mau" quota toggle tip

This pr includes following updates:
- Rename `PlanQuotaTable` to `PlanComparisonTable` since the new name is more accurate
- Refactor the original `PlanQuotaTable` component to allow us customizing data cell content, since the original implementation is rigescent and make us difficult to add tips or other information for the data cell, so I setup render functions for each quota item.

## Testing

image

image

image

image

image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/consts/plan-quotas.ts [hunk 1]
```diff
@@ -9,6 +10,8 @@ import {
   type SubscriptionPlanQuota,
 } from '@/types/subscriptions';
 
+import { isDevFeaturesEnabled as isDevelopmentFeaturesEnabled } from './env';
```
Comment (charIeszhao) [inline packages/console/src/consts/plan-quotas.ts:13]: nit: no need to rename?

File: packages/console/src/consts/plan-quotas.ts [hunk 2]
```diff
@@ -96,7 +106,7 @@ const enterprisePlanTable: SubscriptionPlanTable = {
   appLogoAndFaviconEnabled: true,
   darkModeEnabled: true,
   i18nEnabled: true,
-  mfaEnabled: true,
+  mfaEnabled: undefined,
```
Comment (charIeszhao) [inline packages/console/src/consts/plan-quotas.ts:109]: Should we use `isDevFeaturesEnabled` to keep the backward compatibility?

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/PlanQuotaKeyLabel/index.tsx [hunk 3]
```diff
@@ -37,15 +42,23 @@ const planQuotaKeyPhraseMap: {
...
 
 type Props = {
   quotaKey: keyof SubscriptionPlanTable;
 };
 
 function PlanQuotaKeyLabel({ quotaKey }: Props) {
-  return <DynamicT forKey={`subscription.quota_table.${planQuotaKeyPhraseMap[quotaKey]}`} />;
+  const phraseKey = planQuotaKeyPhraseMap[quotaKey];
+  return (
+    cond(phraseKey && <DynamicT forKey={`subscription.quota_table.${phraseKey}`} />) ?? <div>-</div>
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/PlanQuotaKeyLabel/index.tsx:60]: The `div` might not be necessary?

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/BetaFeatureFlag.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { cond, type Optional } from '@silverhand/essentials';
+
+import DynamicT from '@/ds-components/DynamicT';
+
+import QuotaValueWrapper from './QuotaValueWrapper';
+
+type Props = {
+  isEnabled: Optional<boolean>;
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/BetaFeatureFlag.tsx:8]: nit: `isEnabled?: boolean` for simplicity

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/GeneralFeatureFlag.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import Success from '@/assets/icons/success.svg';
+import DynamicT from '@/ds-components/DynamicT';
+
+import QuotaValueWrapper from './QuotaValueWrapper';
+
+type Props = {
+  isEnabled: Optional<boolean>;
+};
+
+function GeneralFeatureFlag({ isEnabled }: Props) {
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/GeneralFeatureFlag.tsx:12]: What is `general feature`? Maybe `generic`?

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauLimit.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { type Optional, type Nullable, cond } from '@silverhand/essentials';
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauLimit.tsx:1]: It seems these `FooLimit` components are quite similar. Can we abstract one component to render them all?

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauUnitPrices.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  return prices.length === 0 ? (
+    <div>-</div>
+  ) : (
+    <div>
+      {prices.map((value, index) => (
+        <div key={value}>
+          <DynamicT forKey="subscription.quota_table.tier" interpolation={{ value: index + 1 }} />
+          <DynamicT
+            forKey="subscription.quota_table.mau_price"
+            interpolation={{ value: (Number(value) / 100).toFixed(3) }}
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauUnitPrices.tsx:29]: Why `toFixed(3)`?

**Review by charIeszhao — APPROVED** [review]
Overall looks good to me.

**Comment by charIeszhao** [comment]
nit:

image
The wording needs to be revised. Reads a bit odd.

File: packages/console/src/consts/plan-quotas.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/consts/plan-quotas.ts:13]: It's changed by the lint rule, will update the rule in another PR later.

File: packages/console/src/consts/plan-quotas.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/consts/plan-quotas.ts]: Yes, updated.

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/PlanQuotaKeyLabel/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/PlanQuotaKeyLabel/index.tsx]: It's needed, but updated with react fragment.

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/BetaFeatureFlag.tsx (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/BetaFeatureFlag.tsx]: Updated.

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/GeneralFeatureFlag.tsx (hunk 5 above)
Comment (xiaoyijun) [reply packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/GeneralFeatureFlag.tsx]: Yes, use `generic` instead.

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauLimit.tsx (hunk 6 above)
Comment (xiaoyijun) [reply packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauLimit.tsx]: Updated.

File: packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauUnitPrices.tsx (hunk 7 above)
Comment (xiaoyijun) [reply packages/console/src/pages/TenantSettings/Subscription/PlanComparisonTable/renderers/MauUnitPrices.tsx:28]: Yeah, it's redundant, removed.

**Pull request merged**